pkg/actions/karpenter/create_test.go

```diff
@@ -0,0 +1,219 @@
+)
+
+var _ = Describe("Create", func() {
+	Context("Create Karpenter Installation", func() {
+		var (
+			clusterName            string
+			p                      *mockprovider.MockProvider
+			cfg                    *api.ClusterConfig
+			fakeStackManager       *managerfakes.FakeStackManager
+			ctl                    *eks.ClusterProvider
+			fakeKarpenterInstaller *karpenterfakes.FakeInstallKarpenter
+		)
+
+		BeforeEach(func() {
+			clusterName = "my-cluster"
+			p = mockprovider.NewMockProvider()
+			cfg = api.NewClusterConfig()
+			cfg.Metadata.Name = clusterName
+			cfg.VPC = vpcConfig()
+			cfg.AvailabilityZones = []string{"us-west-2a", "us-west-2b"}
+			fakeStackManager = &fakes.FakeStackManager{}
+			fakeKarpenterInstaller = &karpenterfakes.FakeInstallKarpenter{}
+			ctl = &eks.ClusterProvider{
+				Provider: p,
+				Status: &eks.ProviderStatus{
+					ClusterInfo: &eks.ClusterInfo{
+						Cluster: testutils.NewFakeCluster(clusterName, ""),
+					},
+				},
+			}
+			fakeStackManager.CreateStackStub = func(_ string, rs builder.ResourceSet, _ map[string]string, _ map[string]string, errs chan error) error {
+				go func() {
+					errs <- nil
+				}()
+				return nil
+			}
+			p.MockEC2().On("DescribeSubnets", &ec2.DescribeSubnetsInput{
+				SubnetIds: aws.StringSlice([]string{
+					privateSubnet1,
+					privateSubnet2,
+					publicSubnet1,
+					publicSubnet2,
+				}),
+			}).Return(&ec2.DescribeSubnetsOutput{
+				Subnets: []*ec2.Subnet{
+					{
+						SubnetId: aws.String(privateSubnet1),
+						VpcId:    aws.String(cfg.VPC.ID),
+					},
+					{
+						SubnetId: aws.String(privateSubnet2),
+						VpcId:    aws.String(cfg.VPC.ID),
+						Tags: []*ec2.Tag{
+							{
+								Key:   aws.String("kubernetes.io/cluster/" + clusterName),
+								Value: aws.String("shared"),
+							},
+						},
+					},
+					{
+						SubnetId: aws.String(publicSubnet1),
+						VpcId:    aws.String(cfg.VPC.ID),
+					},
+					{
+						SubnetId: aws.String(publicSubnet2),
+						VpcId:    aws.String(cfg.VPC.ID),
+					},
+				},
+			}, nil)
+
+			p.MockEC2().On("CreateTags", &ec2.CreateTagsInput{
+				Resources: []*string{
+					&privateSubnet1,
+					&publicSubnet1,
+					&publicSubnet2,
+				},
+				Tags: []*ec2.Tag{
+					{
+						Key:   aws.String("kubernetes.io/cluster/" + clusterName),
+						Value: aws.String(""),
+					},
+				},
+			}).Return(&ec2.CreateTagsOutput{}, nil)
+		})
+		It("can install Karpenter on an existing cluster", func() {
+			fakeKarpenterInstaller.InstallReturns(nil)
+			install := &karpenteractions.Installer{
+				StackManager:       fakeStackManager,
+				CTL:                ctl,
+				Config:             cfg,
+				KarpenterInstaller: fakeKarpenterInstaller,
+			}
+			Expect(install.Create()).To(Succeed())
+			Expect(fakeKarpenterInstaller.InstallCallCount()).To(Equal(1))
+		})
+		FWhen("DescribeSubnets fails", func() {
+			BeforeEach(func() {
+				p.MockEC2().On("DescribeSubnets", mock.Anything).Return(nil, errors.New("nope"))
```

Comment:
@aclevername HALP :D This mock is not working for some reason, I don't understand. :D See anything wrong with it?